Zend/zend.c

```diff
@@ -177,11 +177,25 @@ static ZEND_INI_MH(OnSetExceptionStringParamMaxLen) /* {{{ */
 
 static ZEND_INI_MH(OnUpdateFiberStackSize) /* {{{ */
 {
+	zend_long size;
+
 	if (new_value) {
-		EG(fiber_stack_size) = zend_atol(ZSTR_VAL(new_value), ZSTR_LEN(new_value));
+		size = zend_atol(ZSTR_VAL(new_value), ZSTR_LEN(new_value));
+		if (size == 0) {
+			size = ZEND_FIBER_DEFAULT_C_STACK_SIZE;
+		} else if (size < ZEND_FIBER_MIN_C_STACK_SIZE) {
+			size = ZEND_FIBER_MIN_C_STACK_SIZE;
+		} else if (size > ZEND_FIBER_MAX_C_STACK_SIZE) {
```

Comment:
Do we actually need a maximum here? Does it pose any issues if exceeded, or is it arbitrary?

Reply:
I do not think it's a good idea to set a huge C stack size for fibers, it is usually a mistake to set it a huge value.
Professionals can modify ZEND_FIBER_MAX_C_STACK_SIZE to remove the limitation, it is also a kind of configuration.